### ISSUE TITLE
Expand index.codec docs to mention DEFLATE

### ISSUE THREAD

**Issue by markwalkom** [elastic/elasticsearch#13541]
Given we use DEFLATE as this enhanced compression method, as per [here](https://www.elastic.co/blog/store-compression-in-lucene-and-elasticsearch), we should mention it :)

**Comment by rmuir** [comment]
Its already documented: https://www.elastic.co/guide/en/elasticsearch/reference/master/index-modules.html
```
index.codec
    [experimental] This functionality is experimental and may be changed or removed completely in a
 future release. The default value compresses stored data with LZ4 compression, but this can be set 
to best_compression for a higher compression ratio, at the expense of slower stored fields performance. 
```

I don't see what else needs to be done here?

**Comment by s1monw** [comment]
I don't think we need to document the actual algorithm we use since it's an impl detail?

**Comment by markwalkom** [comment]
It's an easy change. It makes things explicit.
Given it took less words than all the above there is now a PR.

**Comment by rmuir** [comment]
I don't mind the change, its just that its more complicated than that. Actually deflate vs lz4 matters very little: we've had experimental deflate in lucene since... 4.1 i think? But it would not actually compress any better in most cases than lz4, even though it was set to level 9!

Currently we use a hack, a huge block size (60KB vs 16KB) to allow this algorithm to compress better within the confines of the current format.

But what is really needed for it to kick ass is to have shared dictionary support. Its just nontrivial to have both shared dictionary support *and* bulk merging without recompression *and* low memory overhead *and* good performance and all this shit that people want.

Also the java apis around deflate dictionaries are weak: you don't have access to the current compression dictionary like the trick used for http://zlib.net/pigz/ or any of that: so I wouldnt be surprised if the algorithm was changed to something else in the future.

Anyway full details are already in the lucene file formats docs (http://lucene.apache.org/core/5_3_0/core/org/apache/lucene/codecs/lucene50/Lucene50StoredFieldsFormat.html)

**Comment by markwalkom** [comment]
Does the same comments above (roughly) apply for LZ4?

**Comment by clintongormley** [comment]
Closed by #13549
